src/Logger/BanditActionEvent.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+namespace Eppo\Logger;
+
+use DateTime;
+use Eppo\DTO\Bandit\Bandit;
+use Eppo\DTO\Bandit\BanditEvaluation;
+use JsonSerializable;
+use Serializable;
+
+class BanditActionEvent implements Serializable, JsonSerializable
+{
+    public function __construct(
+        public readonly string $flagKey,
+        public readonly string $banditKey,
+        public readonly string $subjectKey,
+        public readonly ?string $action,
+        public readonly ?float $actionProbability,
+        public readonly ?float $optimalityGap,
+        public readonly ?string $modelVersion,
+        public readonly DateTime $timestamp,
+        public readonly array $subjectNumericAttributes,
+        public readonly array $subjectCategoricalAttributes,
+        public readonly array $actionNumericAttributes,
+        public readonly array $actionCategoricalAttributes,
+        public readonly array $metaData
+    ) {
+    }
+
+    public static function fromEvaluation(
+        string $banditKey,
+        BanditEvaluation $result,
+        Bandit $bandit,
+        $metaData
+    ): BanditActionEvent {
+        return new self(
+            $result->flagKey,
+            $banditKey,
+            $result->subjectKey,
+            $result->selectedAction,
+            $result->actionWeight,
+            $result->optimalityGap,
+            $bandit->modelVersion,
+            new DateTime(),
+            $result->subjectAttributes->numericAttributes,
+            $result->subjectAttributes->categoricalAttributes,
+            $result->actionAttributes->numericAttributes,
+            $result->actionAttributes->categoricalAttributes,
+            $metaData
+        );
+    }
+
+    public function serialize(): string
+    {
+        return serialize($this->__serialize());
+    }
+
+    public function unserialize(string $data): void
+    {
+        $unserializedData = unserialize($data);
+        $this->__unserialize($unserializedData);
+    }
+
+    public function __serialize(): array
+    {
+        return [
+            'flagKey' => $this->flagKey,
+            'banditKey' => $this->banditKey,
+            'subjectKey' => $this->subjectKey,
+            'action' => $this->action,
+            'actionProbability' => $this->actionProbability,
+            'optimalityGap' => $this->optimalityGap,
+            'modelVersion' => $this->modelVersion,
+            'timestamp' => $this->timestamp->format('Y-m-d\TH:i:sP'), // Convert DateTime to a serializable format
```

Comment:
should we include fractions of a second?

Reply:
I've changed this to `microtime(true)` call, and for parity changed it in AssignmentEvent as well